rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -419,10 +447,35 @@ class TimeSource::NodeState final
     clock_subscription_.reset();
   }
 
+  // On set Parameters callback handle
+  node_interfaces::OnSetParametersCallbackHandle::SharedPtr on_set_parameters_callback_{nullptr};
+
   // Parameter Event subscription
   using ParamSubscriptionT = rclcpp::Subscription<rcl_interfaces::msg::ParameterEvent>;
   std::shared_ptr<ParamSubscriptionT> parameter_subscription_;
 
+  // Callback for parameter settings
+  rcl_interfaces::msg::SetParametersResult on_set_parameters(
```

Comment:
the `on_parameter_event` function below does
```
      if (it.second->value.type != ParameterType::PARAMETER_BOOL) {
        RCLCPP_ERROR(logger_, "use_sim_time parameter cannot be set to anything but a bool");
        continue;
      }
```

should maybe this check on the type be moved here? we should fail the "parameter set" if someone uses the wrong type.

Reply:
I was going to move the check up, but c54a6f1cd2b5bfbe530c362ad7e8fa22f753e325 gave a good reasoning for why the check was removed from the old `on_set_parameters_callback` .